test/async-hooks/test-async-exec-resource-http-agent.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+const {
+  executionAsyncResource,
+  executionAsyncId,
+  createHook,
+} = require('async_hooks');
+const http = require('http');
+
+const hooked = {};
+createHook({
+  init(asyncId, type, triggerAsyncId, resource) {
+    hooked[asyncId] = resource;
+  }
+}).enable();
+
+const agent = new http.Agent({
+  maxSockets: 1
+});
+
+const server = http.createServer((req, res) => {
+  res.end('ok');
+});
+
+server.listen(0, () => {
+  assert.strictEqual(executionAsyncResource(), hooked[executionAsyncId()]);
+
+  http.get({ agent, port: server.address().port }, () => {
```

Comment:
Can you wrap all the callbacks that are supposed to be called exactly once in `common.mustCall()`, i.e. the ones to `http.createServer()`, `server.listen()` and `http.get()`? That ensures that the assertions here are actually run, instead of the test just passing because something led to the callbacks not being called. (That’s probably not going to realistically happen because a ton of other tests would fail is these things were broken, but it’s a good practice to add those checks in our tests.)